OpenSim/Simulation/MarkersReference.cpp

```diff
@@ -24,133 +24,168 @@
 #include "MarkersReference.h"
 #include <OpenSim/Common/Units.h>
 #include <SimTKcommon/internal/State.h>
+#include <cmath>
 
 using namespace std;
 using namespace SimTK;
 
 namespace OpenSim {
 
-MarkersReference::MarkersReference() : Reference_<SimTK::Vec3>()
-{
+MarkersReference::MarkersReference() :
+    Reference_<SimTK::Vec3>() {
     constructProperties();
     setAuthors("Ajay Seth");
 }
 
 MarkersReference::MarkersReference(const std::string& markerFile,
-    Units modelUnits) : MarkersReference()
-{
+                                   Units modelUnits) :
+    MarkersReference() {
     loadMarkersFile(markerFile, modelUnits);
 }
 
-
-MarkersReference::MarkersReference(MarkerData *markerData,
-    const Set<MarkerWeight>* markerWeightSet) : MarkersReference()
-{
-    _markerData.reset(markerData);
-    if (markerWeightSet!=nullptr) upd_marker_weights()= *markerWeightSet;
-    populateFromMarkerData(*markerData);
+MarkersReference::
+MarkersReference(const TimeSeriesTable_<SimTK::Vec3>& markerTable,
```

Comment:
If you use this constructor with a Table you read in with  TRCFileAdapter we will not have the Units reflected when asking for the reference value, which is assumed to be in SI units.

As a *note*, a `TableSource` created from a TRC file would have the same issue, where the column output values would be in *mm*, for example, but all the tools, algorithms expect *m*.